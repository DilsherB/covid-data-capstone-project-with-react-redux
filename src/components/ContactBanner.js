import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import {
  FaLinkedin,
  FaFacebook,
  FaGithub,
  FaStackOverflow,
} from "react-icons/fa";
import Photo from "../assets/My Profile Photo.jpeg";

const ContactBanner = () => {
  return (
    <div className="bg-slate-300 rounded-lg p-3 flex gap-6 mb-5">
      <img alt="Developer" src={Photo} className="rounded-full w-28 h-28" />
      <div>
        <h1 className="text-2xl font-bold">Dilsher Balouch</h1>
        <p className="text-lg font-semibold text-slate-600">
          Full-Stack Developer
        </p>
        <p className="text-lg font-semibold text-slate-600 flex justify-between">
          <a
            href="https://www.linkedin.com/in/dilsherbalouch/"
            target="_blank"
            rel="noreferrer"
          >
            <FaLinkedin />
          </a>
          <a
            href="https://www.facebook.com/dilsherb/"
            target="_blank"
            rel="noreferrer"
          >
            <FaFacebook />
          </a>
          <a
            href="https://github.com/DilsherB"
            target="_blank"
            rel="noreferrer"
          >
            <FaGithub />
          </a>
          <a
            href="https://stackoverflow.com/users/17731460/dilsher-balouch"
            target="_blank"
            rel="noreferrer"
          >
            <FaStackOverflow />
          </a>
        </p>
      </div>
    </div>
  );
};

export default ContactBanner;
