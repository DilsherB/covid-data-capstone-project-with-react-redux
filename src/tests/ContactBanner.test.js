import renderer from "react-test-renderer";
import React from "react";
import ContactBanner from "../components/ContactBanner";

jest.mock("react-icons/fa");
jest.mock("../assets/My Profile Photo.jpeg");

const renderTree = (tree) => renderer.create(tree);
describe("<ContactBanner>", () => {
  it("should render component", () => {
    expect(renderTree(<ContactBanner />).toJSON()).toMatchSnapshot();
  });
});
