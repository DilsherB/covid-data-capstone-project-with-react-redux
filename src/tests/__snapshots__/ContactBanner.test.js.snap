// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactBanner> should render component 1`] = `
<div
  className="bg-slate-300 rounded-lg p-3 flex gap-6 mb-5"
>
  <img
    alt="Developer"
    className="rounded-full w-28 h-28"
    src="My Profile Photo.jpeg"
  />
  <div>
    <h1
      className="text-2xl font-bold"
    >
      Dilsher Balouch
    </h1>
    <p
      className="text-lg font-semibold text-slate-600"
    >
      Full-Stack Developer
    </p>
    <p
      className="text-lg font-semibold text-slate-600 flex justify-between"
    >
      <a
        href="https://www.linkedin.com/in/dilsherbalouch/"
        rel="noreferrer"
        target="_blank"
      />
      <a
        href="https://www.facebook.com/dilsherb/"
        rel="noreferrer"
        target="_blank"
      />
      <a
        href="https://github.com/DilsherB"
        rel="noreferrer"
        target="_blank"
      />
      <a
        href="https://stackoverflow.com/users/17731460/dilsher-balouch"
        rel="noreferrer"
        target="_blank"
      />
    </p>
  </div>
</div>
`;
